src/main/java/jdbchomework/dao/jdbc/AbstractDao.java

```diff
@@ -2,118 +2,116 @@
-        try {
-            transaction = session.beginTransaction();
-            result = session.createQuery("FROM " + entity).list();
-            transaction.commit();
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (Statement statement = connection.createStatement()) {
+            String sql = "SELECT * FROM " + table;
+            ResultSet resultSet = statement.executeQuery(sql);
+            while (resultSet.next()) {
+                T t = createEntity(resultSet);
+                result.add(t);
             }
-            log.error(CANNOT_CONNECT_TO_DB, e);
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
         }
         return result;
     }
 
 
     public T getById(long id) {
         T result = null;
-        Session session = getSession();
-        Transaction transaction = null;
-        try {
-            transaction = session.beginTransaction();
-            result = session.get(clazz, id);
-            transaction.commit();
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (
+                PreparedStatement statement = connection
+                        .prepareStatement("SELECT * FROM " + table + " WHERE " + column + " = ?;")) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+            if (rs.next()) {
+                result = createEntity(rs);
             }
-            log.error(CANNOT_CONNECT_TO_DB, e);
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
         }
         return result;
     }
 
-    public boolean updateById(long id, T toUpdate) {
+    public boolean updateById( long id, T toUpdate) {
         boolean result = false;
-        Session session = getSession();
-        Transaction transaction = null;
-        try {
-            transaction = session.beginTransaction();
-            toUpdate.setId(id);
-            session.update(toUpdate);
-            transaction.commit();
-        } catch (HibernateException e) {
-            if (transaction != null) {
-                transaction.rollback();
+        try (PreparedStatement statement = connection
+                .prepareStatement("UPDATE " + table + " SET name = ? WHERE " + column + " =?;")) {
+            String name = toUpdate.getName();
+            statement.setString(1, name);
+            statement.setLong(2, id);
+            if (statement.executeUpdate() > 0) {
+                result = true;
             }
-            log.error(CANNOT_CONNECT_TO_DB, e);
+        } catch (SQLException e) {
+            log.error("Cannot connect to DB", e);
+            throw new RuntimeException("Cannot connect to DB", e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)